### PR TITLE
[tests-only] Run pipeline steps in parallel

### PR DESCRIPTION
**Pull request by saw-jan**
In this PR, I have added some helper functions to add pipeline dependencies for pipelines. Also, made some steps to run in parallel and some dependent to others.
For e.g;
`submodules`, `install-core` and `generate` steps can run in parallel.

Closes https://github.com/owncloud/client/issues/9557

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/11680/6/1
:boom: The GUI tests failed.
GUI Logs: (https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/index.html)
Server Logs: (https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/serverlog.log)
Screenshots:
- [Both_original_and_copied_folders_can_be_synced.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Both_original_and_copied_folders_can_be_synced.png)
- [Connect_account_with_manual_sync_folder_option.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Connect_account_with_manual_sync_folder_option.png)
- [File_with_long_name_(>228_characters)_is_blacklisted.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/File_with_long_name_(>228_characters)_is_blacklisted.png)
- [File_with_long_name_can_be_synced.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/File_with_long_name_can_be_synced.png)
- [File_with_spaces_in_the_name_can_sync.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/File_with_spaces_in_the_name_can_sync.png)
- [Filenames_that_are_rejected_by_the_server_are_reported.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Filenames_that_are_rejected_by_the_server_are_reported.png)
- [Invalid_system_names_are_synced_in_linux.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Invalid_system_names_are_synced_in_linux.png)
- [Many_subfolders_can_be_synced.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Many_subfolders_can_be_synced.png)
- [Sync_all_is_selected_by_default.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Sync_all_is_selected_by_default.png)
- [Sync_only_one_folder_from_the_server.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Sync_only_one_folder_from_the_server.png)
- [Syncing_a_file_from_the_server_and_creating_a_conflict.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Syncing_a_file_from_the_server_and_creating_a_conflict.png)
- [Syncing_a_folder_to_the_server.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Syncing_a_folder_to_the_server.png)
- [Syncing_file_of_1_GB_size.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Syncing_file_of_1_GB_size.png)
- [Syncing_folders_each_having_500_files.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Syncing_folders_each_having_500_files.png)
- [Verify_one_empty_folder_with_a_length_longer_than_the_allowed_limit_will_not_be_synced.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Verify_one_empty_folder_with_a_length_longer_than_the_allowed_limit_will_not_be_synced.png)
- [Verify_pre_existing_folders_in_local_(Desktop_client)_are_copied_over_to_the_server.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Verify_pre_existing_folders_in_local_(Desktop_client)_are_copied_over_to_the_server.png)
- [Verify_that_you_can_create_a_subfolder_with_long_name.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/Verify_that_you_can_create_a_subfolder_with_long_name.png)
- [sort_folders_list_by_name_and_size.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/sort_folders_list_by_name_and_size.png)
- [various_types_of_files_can_be_synced_from_client_to_server.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/various_types_of_files_can_be_synced_from_client_to_server.png)
- [various_types_of_files_can_be_synced_from_server_to_client.png](https://cache.owncloud.com/public/owncloud/client/11680/guiReportUpload/screenshots/various_types_of_files_can_be_synced_from_server_to_client.png)

File: .drone.star [hunk 1]
```diff
@@ -163,52 +118,29 @@ def check_starlark(ctx, trigger = {}, depends_on = []):
                 },
             },
         ],
-        "depends_on": depends_on,
         "trigger": trigger,
-    }
+    }]
 
-def build_and_test_client(ctx, c_compiler, cxx_compiler, build_type, generator, trigger = {}, depends_on = []):
+def ctest_pipeline(ctx, c_compiler, cxx_compiler, build_type, generator, trigger = {}):
```
Comment (Talank) [inline .drone.star:124]: I think unit_test_pipeline will be nice name to this method, as it does unit tests

**Review by Talank — COMMENTED** [review]
looks good

File: .drone.star [hunk 2]
```diff
@@ -267,19 +191,18 @@ def gui_tests(ctx, trigger = {}, depends_on = [], filterTags = [], version = "da
...
         ],
-    }
+    }]
 
-def build_client(ctx, c_compiler, cxx_compiler, build_type, generator, build_command, build_dir):
+def build_client(c_compiler, cxx_compiler, build_type, generator, build_command, build_dir):
     return [
         {
-            "name": "cmake",
+            "name": "generate",
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply .drone.star:205]: I just wanted to use a more generic name

**Review by individual-it — APPROVED** [review]

**Pull request merged**